libs/api.lua

```diff
@@ -136,29 +200,32 @@ return function (prefix)
       return next(authors) and authors
     end,
     "^/search/(.*)$", function (query)
+      -- escape all special pattern characters except . and *
+      query = query:gsub("[%(%)%%%+%-%?%[%]%^%$]", function(c) return "%" .. c end)
       local matches = {}
+      local authorQuery, separator, packageQuery = query:match("^(.-)(/?)([^/]*)$")
```

Comment:
What about deep package names like `foo/bar/baz`?  I don't think any actually exist, but I've tried to keep that code path supported.